File: Day 01/Day1.py
import os

def load_input(file):
    with open(file, 'r') as lines:
        return [line for line in lines]

def sum_calories(calorie_list):
    calories = []
    total = 0
    for line in calorie_list:
        try:
            total = total + int(line)
        except:
            calories.append(total)
            total=0
    return calories

def main():
    file_name = os.path.join(os.path.dirname(__file__), "input.txt")
    calorie_list = load_input(file_name)
    
    calories = sum_calories(calorie_list)
    calories.sort(reverse=True)
    print("Part 1: " + str(calories[0]))
        
    top3 = calories[0] + calories[1] + calories[2]
    print("Part 2: " + str(top3))

if __name__ == '__main__':
    main()

File: Day 01/Day1.cpp
#include <iostream>
#include <fstream>
#include <bits/stdc++.h>

using namespace std;

int main()
{
    ifstream file("input.txt");
    int calories[500]={}; //zakladamy "na oko", ze nie bedzie wiecej elfow
    int total = 0;
    int elf = 0;
    string line;

    //wczytanie i zsumowanie kalorii kazdego elfa
    while(!file.eof()){
        getline(file, line);
        if(line == ""){
            calories[elf] = total;
            elf++;
            total=0;
        }
        else
            total += stoi(line); //zamiana string -> int
    }
    calories[elf] = total;

    sort(calories, calories+elf+1, greater<int>());

    //for(int i=0; i<elf+1; i++)
    //    cout<<i<<" "<<calories[i]<<endl;

    cout<<"Part 1: "<<calories[0]<<endl;
    int top3 = calories[0]+calories[1]+calories[2];
    cout<<"Part 2: "<<top3<<endl;

}
